### ISSUE TITLE
[bug] Transcode error on linux/arm docker build

### ISSUE THREAD

**Issue by smendig** [jellyfin/jellyfin#2126]
**Describe the bug**

Unable to transcode using the latest [docker build (v10.4.3) for linux/arm](https://hub.docker.com/layers/jellyfin/jellyfin/10.4.3/images/sha256-071fd284fd89fd0200bf6832191650815931439f4cf9b43b3dc7913f538d5ac9), from official Jellyfin docker hub. The error happens for any transcode on default configuration.

**To Reproduce**
1. Run official docker image jellyfin/jellyfin:10.4.3-arm with no previous configuration (no /config volume)
2. Add any media library
3. Play any item from media library and force transcode (by selecting a different quality option for example)
4. See error: "Playback Error, PlaybackErrorNoCompatibleStream"

**Logs**
```[INF] Current HLS implementation doesn't support non-keyframe breaks but one is requested, ignoring that request
 [INF] /usr/bin/ffmpeg -f mov,mp4,m4a,3gp,3g2,mj2 -i file:"/media/movies/Giant (2017)/Giant.2017.mp4" -map_metadata -1 -map_chapters -1 -threads 0 -map 0:0 -map 0:1 -map -0:s -codec:v:0 libx264 -pix_fmt yuv420p -preset veryfast -crf 23 -maxrate 864691 -bufsize 1729382 -profile:v high -level 4.1 -x264opts:0 subme=0:me_range=4:rc_lookahead=10:me=dia:no_chroma_me:8x8dct=0:partitions=none -force_key_frames:0 "expr:gte(t,0+n_forced*3)" -vf "scale=trunc(min(max(iw\,ih*dar)\,640)/2)*2:trunc(ow/dar/2)*2" -copyts -vsync -1 -codec:a:0 libmp3lame -ac 2 -ab 135310  -f hls -max_delay 5000000 -avoid_negative_ts disabled -start_at_zero -hls_time 3 -individual_header_trailer 0 -hls_segment_type mpegts -start_number 0 -hls_segment_filename "/config/transcoding-temp/a497d36053f742771940e8b257333d69%d.ts" -hls_playlist_type vod -hls_list_size 0 -y "/config/transcoding-temp/a497d36053f742771940e8b257333d69.m3u8"
 [ERR] FFMpeg exited with code 1
 [WRN] cannot serve /config/transcoding-temp/a497d36053f742771940e8b257333d690.ts as transcoding quit before we got there
 [ERR] Error processing request: Could not find file '/config/transcoding-temp/a497d36053f742771940e8b257333d690.ts'.
```

Attaching into container to execute the same command to view the error:
```
/usr/bin/ffmpeg -f mov,mp4,m4a,3gp,3g2,mj2 -i file:"/media/movies/Giant (2017)/Giant.2017.mp4" -map_metadata -1 -map_chapters -1 -threads 0 -map 0:0 -map 0:1 -map -0:s -codec:v:0 libx264 -pix_fmt yuv420p -preset veryfast -crf 23 -maxrate 864691 -bufsize 1729382 -profile:v high -level 4.1 -x264opts:0 subme=0:me_range=4:rc_lookahead=10:me=dia:no_chroma_me:8x8dct=0:partitions=none -force_key_frames:0 "expr:gte(t,0+n_forced*3)" -vf "scale=trunc(min(max(iw\,ih*dar)\,640)/2)*2:trunc(ow/dar/2)*2" -copyts -vsync -1 -codec:a:0 libmp3lame -ac 2 -ab 135310  -f hls -max_delay 5000000 -avoid_negative_ts disabled -start_at_zero -hls_time 3 -individual_header_trailer 0 -hls_segment_type mpegts -start_number 0 -hls_segment_filename "/config/transcoding-temp/a497d36053f742771940e8b257333d69%d.ts" -hls_playlist_type vod -hls_list_size 0 -y "/config/transcoding-temp/a497d36053f742771940e8b257333d69.m3u8"
ffmpeg version 3.2.14-1~deb9u1 Copyright (c) 2000-2019 the FFmpeg developers
  built with gcc 6.3.0 (Debian 6.3.0-18+deb9u1) 20170516
  configuration: --prefix=/usr --extra-version='1~deb9u1' --toolchain=hardened --libdir=/usr/lib/arm-linux-gnueabihf --incdir=/usr/include/arm-linux-gnueabihf --enable-gpl --disable-stripping --enable-avresample --enable-avisynth --enable-gnutls --enable-ladspa --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libebur128 --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libmp3lame --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-omx --enable-openal --enable-opengl --enable-sdl2 --enable-libdc1394 --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libopencv --enable-libx264 --enable-shared
  libavutil      55. 34.101 / 55. 34.101
  libavcodec     57. 64.101 / 57. 64.101
  libavformat    57. 56.101 / 57. 56.101
  libavdevice    57.  1.100 / 57.  1.100
  libavfilter     6. 65.100 /  6. 65.100
  libavresample   3.  1.  0 /  3.  1.  0
  libswscale      4.  2.100 /  4.  2.100
  libswresample   2.  3.100 /  2.  3.100
  libpostproc    54.  1.100 / 54.  1.100
Unrecognized option 'hls_segment_type'.
Error splitting the argument list: Option not found
```

**System (please complete the following information):**
 - OS: Docker (linux/arm, build v10.4.3)
 - Browser: Any
 - Jellyfin Version: 10.4.3
 - Installed Plugins: none
 - Reverse proxy: none

**Additional context**

[Default Jellyfin build](https://hub.docker.com/layers/jellyfin/jellyfin/10.4.3/images/sha256-3af2de1bc55f2c7aecb3b8699b7d205d6d91efa8324cf4c0b12cba2b616e9313) (jellyfin/jellyfin:latest a.k.a. jellyfin/jellyfin:10.4.3) uses **ffmpeg v4.1** while the ARM variant uses **ffmpeg v3.2**

**Comment by Maiux92** [comment]
Hi,
I also had the same problem (I run jellyfin on a raspberry pi).
Other people have the same problem, and I found on another issue (I don't rember which one) a solution that works for me.

You have to install ffmpeg-jellyfin and then you have to change the path of ffmpeg inside the playback settings.
For doing so you can do this:
```
docker exec -it container_id /bin/bash
apt update && apt install wget
```
Then, you have to download and install jellyfin-ffmpeg from https://repo.jellyfin.org/releases/server/debian/versions/jellyfin-ffmpeg/.
The latest version gives me an error (libmmal_core.so not found). So I use the 4.2.1-2 version.

```
wget https://repo.jellyfin.org/releases/server/debian/versions/jellyfin-ffmpeg/4.2.1-2/jellyfin-ffmpeg_4.2.1-2-stretch_armhf.deb
dpkg -i jellyfin-ffmpeg_4.2.1-2-stretch_armhf.deb
```

Now, log in into your jellyfin interface and go to Dashboard -> Playback.
In FFmpeg path insert (you probably should use the navigation to correctly select it): /usr/share/jellyfin-ffmpeg/ffmpeg

**Comment by jokerknight** [comment]
the same issue on my raspberry pi 4.

**Comment by jokerknight** [comment]
> Hi,
> I also had the same problem (I run jellyfin on a raspberry pi), and I found in another issue a solution (I don't rember which one).
> 
> You have to install ffmpeg-jellyfin and then you have to change the path of ffmpeg inside the playback settings.
> For doing so, you need to open a shell inside the container:
> 
> ```
> docker exec -it container_id /bin/bash
> ```
> 
> Then, you have to download and install jellyfin-ffmpeg from https://repo.jellyfin.org/releases/server/debian/versions/jellyfin-ffmpeg/.
> The latest version gives me an error (libmmal_core.so not found). So I use the 4.2.1-2 version.
> 
> You can do all these steps with the following commands:
> 
> ```
> apt update && apt install wget
> wget https://repo.jellyfin.org/releases/server/debian/versions/jellyfin-ffmpeg/4.2.1-2/jellyfin-ffmpeg_4.2.1-2-stretch_armhf.deb
> dpkg -i jellyfin-ffmpeg_4.2.1-2-stretch_armhf.deb
> ```
> 
> Now, log in into your jellyfin interface and go to Dashboard -> Playback.
> In FFmpeg path insert (you probably should use the navigation to correctly select it): /usr/share/jellyfin-ffmpeg/ffmpeg
> Then, click save and you shoul be good to go.

good resolution,hope official can provide an option or detect local ffmpeg version to fix them.

**Comment by Shlokp** [comment]
@jokerknight Which ffmpeg version did you use for rpi4? There is no ffmpeg file in /usr/share/jellyfin-ffmeg/

I am using the linux/arm build on rpi4

**Comment by basdp** [comment]
Had same problem on Raspberry Pi 4 with Docker latest. This workaround fixed it, but won't survive a reboot, so we need to find a permanent fix.

**Comment by neoranger** [comment]
Excellent Solution!! It's work perfect on my Raspberry Pi 2B+. Thanks!!

**Comment by Strit** [comment]
I just stumpled into this issue aswell.

While the workaround from @Maiux92 does work, it would be great if the ffmpeg in the ARM docker container could be updated to match the x86 version. Especially when the Jellyfin system depends on a newer version.

**Comment by cromefire** [comment]
Maybe someone should bump the Debian version of the container.

**Comment by antipatico** [comment]
I'm using my own Dockerfile on raspberry pi 3.
Note that the ssl and the discovery port are **not** exposed.

```
FROM debian

ENV DEBIAN_FRONTEND="noninteractive"

RUN apt-get update -y && \
    apt-get upgrade -y

RUN apt-get install gnupg wget apt-transport-https apt-utils lsb-release -y

RUN  wget -O /tmp/jellyfin_team.gpg.key https://repo.jellyfin.org/debian/jellyfin_team.gpg.key  && \                                                                                         
    apt-key add /tmp/jellyfin_team.gpg.key && rm -f /tmp/jellyfin_team.gpg.key && \
    echo "deb [arch=$( dpkg --print-architecture )] https://repo.jellyfin.org/debian $( lsb_release -c -s ) main" | tee /etc/apt/sources.list.d/jellyfin.list && \                           
    apt-get update -y && \
    apt-get install jellyfin -y

# Fix bugged jellyfin-ffmpeg as reported in https://github.com/jellyfin/jellyfin/issues/2126

ADD jellyfin-ffmpeg_4.2.1-2-buster_armhf.deb /tmp/

RUN dpkg -i /tmp/jellyfin-ffmpeg_4.2.1-2-buster_armhf.deb && \
    rm -f /tmp/jellyfin-ffmpeg_4.2.1-2-buster_armhf.deb

USER jellyfin:jellyfin

EXPOSE 8096/tcp

CMD ["/usr/bin/jellyfin", "-d", "/var/lib/jellyfin", "-C", \
     "/var/cache/jellyfin", "-c", "/etc/jellyfin", "-l", \
     "/var/log/jellyfin", "--ffmpeg", "/usr/lib/jellyfin-ffmpeg/ffmpeg" ]
```

**Comment by giejay** [comment]
@antipatico Thanks! Where did you find the jellyfin-ffmpeg.deb?

**Comment by JustAMan** [comment]
Eh, if you've added our repository and are installing via `apt install jellyfin` why don't you just do `apt install jellyfin jellyfin-ffmpeg`?

**Comment by giejay** [comment]
Yeah, could have done that as well. For now I'm just using the docker image provided by Linux server, that one is using jellyfin-ffmpeg by default and hardware acceleration is working out of the box.

**Comment by maxirus** [comment]
Did not work for me. I get the following error in the log:

`[20:50:25] [WRN] FFmpeg: Custom: Failed version check: /usr/share/jellyfin-ffmpeg/ffmpeg`

Then I get this error when running a version check:
```sh
# /usr/share/jellyfin-ffmpeg/ffmpeg -version
/usr/share/jellyfin-ffmpeg/ffmpeg: error while loading shared libraries: libmmal_core.so: cannot open shared object file: No such file or directory
```

**Comment by Maiux92** [comment]
> 
> 
> Did not work for me. I get the following error in the log:
> 
> `[20:50:25] [WRN] FFmpeg: Custom: Failed version check: /usr/share/jellyfin-ffmpeg/ffmpeg`
> 
> Then I get this error when running a version check:
> 
> ```shell
> # /usr/share/jellyfin-ffmpeg/ffmpeg -version
> /usr/share/jellyfin-ffmpeg/ffmpeg: error while loading shared libraries: libmmal_core.so: cannot open shared object file: No such file or directory
> ```

I had the same problem on my raspberry pi.

**Comment by Maiux92** [comment]
> 
> 
> Did not work for me. I get the following error in the log:
> 
> `[20:50:25] [WRN] FFmpeg: Custom: Failed version check: /usr/share/jellyfin-ffmpeg/ffmpeg`
> 
> Then I get this error when running a version check:
> 
> ```shell
> # /usr/share/jellyfin-ffmpeg/ffmpeg -version
> /usr/share/jellyfin-ffmpeg/ffmpeg: error while loading shared libraries: libmmal_core.so: cannot open shared object file: No such file or directory
> ```

I had the same problem on my raspberry pi.
I fixed by mapping /opt/vc from the raspberry pi to the container, and by setting the environment variable LD_LIBRARY_PATH=/opt/vc/lib
You probably have to rebuild your container. My docker-compose file is:
```
version: "3"
services:
    jellyfin:
        container_name: jellyfin
        image: jellyfin/jellyfin
        volumes:
            - /media/jellyfin-lib:/media
            - /opt/vc:/opt/vc
        ports:
            - 8096:8096
        restart: unless-stopped
        environment:
            - LD_LIBRARY_PATH=/opt/vc/lib
```

**Comment by jairbubbles** [comment]
With docker image version `10.5.3` it doesn't work by default, however `jellyfin-ffmpeg_4.2.1-5-buster_armhf.deb` is already installed. So the only fix needed is to set the ffmpeg path:

![image](https://user-images.githubusercontent.com/15875066/79140416-426b3c80-7db8-11ea-96da-24943397558e.png)

**Comment by jairbubbles** [comment]
(looks like it's fixed in `10.5.4` with #2779)
